chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,162 @@
+                        |stat_name| get_prometheus_metric_name(stat_name),
+                        stat_name,
+                        value,
+                    )?;
+                }
+            } else {
+                // A summary stats.
+                // A statistic 'a.b.c' creates the following prometheus metrics:
+                // - near_a_b_c_sum
+                // - near_a_b_c_count
+                // - near_a_b_c{quantile="0.95"}
+                for stats_value in values {
+                    match stats_value {
+                        StatsValue::Count(value) => {
+                            self.set_int_value(
+                                |stat_name| get_stats_summary_count_key(stat_name),
+                                |stat_name| get_metric_name_summary_count_gauge(stat_name),
+                                stat_name,
+                                *value,
+                            )?;
+                        }
+                        StatsValue::Sum(value) => {
+                            self.set_int_value(
+                                |stat_name| get_stats_summary_sum_key(stat_name),
+                                |stat_name| get_metric_name_summary_sum_gauge(stat_name),
+                                stat_name,
+                                *value,
+                            )?;
+                        }
+                        StatsValue::Percentile(percentile, value) => {
+                            let key: &str = stat_name;
+                            if !self.gauges.contains_key(key) {
+                                self.gauges.insert(
+                                    key.to_string(),
+                                    try_create_gauge_vec(
+                                        &get_prometheus_metric_name(stat_name),
+                                        stat_name,
+                                        &["quantile"],
+                                    )?,
+                                );
+                            }
+                            self.gauges
+                                .get(key)
+                                .unwrap()
+                                .with_label_values(&[&format!("{:.2}", *percentile as f64 * 0.01)])
+                                .set(*value);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn set_int_value<F1, F2>(
+        &mut self,
+        key_fn: F1,
+        metric_fn: F2,
+        stat_name: &str,
+        value: i64,
+    ) -> anyhow::Result<()>
+    where
+        F1: Fn(&str) -> String,
+        F2: Fn(&str) -> String,
+    {
+        let key: &str = &key_fn(stat_name);
+        if !self.int_gauges.contains_key(key) {
+            self.int_gauges
+                .insert(key.to_string(), try_create_int_gauge(&metric_fn(stat_name), stat_name)?);
+        }
+        self.int_gauges.get(key).unwrap().set(value);
```

Comment:
Here and elsewhere, please use the entry API:

https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.entry

it gets rid of unwraps and cuts down on the number of duplicate map lookiups.

Reply:
Done.
Can't use `or_insert_with()`, because `try_create_int_gauge()` can fail.